perceval/backends/jenkins.py

```diff
@@ -0,0 +1,243 @@
+        """Fetch the builds from the site.
+
+        The method retrieves, from a Jenkins site, the
+        builds updated since the given date.
+
+        :param from_date: Not supported
+
+        :returns: a generator of builds
+        """
+
+        logger.info("Looking for projects at site '%s'", self.site)
+
+        self._purge_cache_queue()
+        nbuilds = 0  # number of builds processed
+        njobs = 0 # number of jobs processed
+
+        if not self.job_url:
+            projects = json.loads(self.client.get_jobs())
+            jobs = projects['jobs']
+        else:
+            jobs = [{'url':self.job_url}]
+
+        for job in jobs:
+            njobs += 1
+            logger.debug("Adding builds from %s (%i/%i)" % (job['url'], njobs, len(jobs)))
+            raw_builds = self.client.get_builds(job['url'])
+            self._push_cache_queue(raw_builds)
+            self._flush_cache_queue()
+            builds = json.loads(raw_builds)
+            builds = builds['builds']
+            for build in builds:
+                yield build
+                nbuilds += 1
+
+        logging.info("Total number of jobs: %i" % (njobs))
+        logging.info("Total number of builds: %i" % (nbuilds))
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the builds from the cache.
+
+        :returns: a generator of builds
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        for items in cache_items:
+            builds = json.loads(items)['builds']
+            for build in builds:
+                yield build
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Build item."""
+        return str(item['url'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a Jenkins item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return float(item['timestamp']/1000)
+
+class JenkinsClient:
+    """Jenkins API client.
+
+    This class implements a simple client to retrieve builds from
+    projects in a Jenkins node.
+
+    :param site: URL of jenkins node: https://build.opnfv.org/ci
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, site):
+        self.site = site
+
+
+    def get_jobs(self):
+        """ Retrieve all jobs
+        """
+        url_jenkins = self.site + "/view/All/api/json"
+
+        req = requests.get(url_jenkins)
+        req.raise_for_status()
+        return req.text
+
+    def get_builds(self, job_url):
```

Comment:
This method is called with the information that comes from the JSON file, the URL for the job, and this URL is complete. So it is better this API, now that it is not used for supporting job_url from command line.